### PR TITLE
Add workflow for automatic LaTeX builds.

### PR DESCRIPTION
**Pull request by hhontheim**
This PR adds a GitHub workflow to the template repository.

On every push to the `master` branch, the workflow builds the LaTeX file, tags the commit, and creates a release on GitHub with the corresponding PDF.

This workflow can also be applied to existing repositories with the commands mentioned in the `README`.

### CODE REVIEW

**Comment by ge65cer** [comment]
@Mtze @krusche Can we merge this please? 🙂

**Comment by Mtze** [comment]
@ge65cer If you review the PR ;)

File: README.md [hunk 1]
```diff
@@ -2,3 +2,20 @@
 A Latex Template for your bachelor / master thesis
 
 Please note: This is only a template. You have to adapt the template to your thesis and discuss the structure of your thesis with your supervisor
+
```
Comment (Mtze) [inline README.md:5]: Please add a heading for this section - I think it is not immediately obvious that this is only important for existing repositories hosted on GitHub

Comment (hhontheim) [reply README.md:5]: Good point! I inserted some headings and also explained how to disable the workflow. I also mentioned the billing aspects for GH Actions briefly.

**Review by ge65cer — APPROVED** [review]

**Review by Mtze — APPROVED** [review]

**Pull request merged**